coding-standards/log-format-mobile-app.md

```diff
@@ -0,0 +1,53 @@
+# Log format
+
+## Requirements
+To create a common log format that fits the most mobile technologies and aids in debugging:
+
+- Include the severity of the log to understand its importance.
+- Include the type of log to segregate between info, error, or warning.
+- Include screen or controller where applicable.
+- Include a unique ID to trace requests across different components or screens.
+- Include source context, which could be a function name, module, or screen name.
+- Include relevant data in JSON format, such as user information, activity details, performance metrics, and data changes.
+- Facilitate debugging of production issues by providing sufficient information in logs, such as stack traces, error messages, and contextual details.
+
+> **Warning**
+> Going against GDPR regulations is a absolute NO
+
+## Proposed solution
+
+``` json
+ {"timestamp": "[timestamp]","level": "[level]","severity": "[severity]","tracebackId": "[unique id]","screenName": "[screen or controller name]","source": "[function/module name]","data": [any relevant data of type JSON],"message": "[log message]","stackTrace": "[stackTrace]"}
+```
+
+### Explanation of each component: 
+
+| Component     | Description                                     |
+| ------------- | ----------------------------------------------- |
+| [timestamp]   | Represents the timestamp when the log entry was created. Use ISO 8601 format, which is widely recognized and provides a standardized representation of date and time (Mandatory) |
+| [level]       | Represents the log level, such as INFO, WARN, ERROR, or FATAL (Mandatory) |
+| [severity]    | Represents the severity level associated with the log entry (e.g., LOW, MEDIUM, HIGH) (Mandatory) |
+| [tracebackId] | A unique identifier associated with the request (Mandatory) |
+| [screenName]  | Name of the screen or controller where the log originated (Optional) |
+| [httpMethod]  | Type of request (Optional) |
```

Comment:
All these would be needed for frontend but we can make **tracebackId** optional.
HttpMethod can have api request type consumed by frontent

Reply:
Hmm I am not sure about this.
If we need httpMethod then we will also need the end point used right, and the data sent in the request etc?

Or even better we keep all these in data property only since this generally applies to API related errors and don't apply for all logs from frontend.

I think we should remove it.

@rajkumar-osmosys  what do you suggest?

Reply:
Yes, I think we can remove it and keep everything in data property only.

Reply:
ok then I'm thinking to remove tracebackId and httpMethod.
Just keep data property (optional) which can have httpmethod, endpoint, and api data
@rajkumar-osmosys

Reply:
updated